packages/react/src/hooks/actions/shared/useTypeCastFields.ts

```diff
@@ -1,16 +1,18 @@
 import * as React from 'react';
+import isString from 'lodash/isString';
 import {
   ModelInit,
+  PersistentModel,
   PersistentModelMetaData,
   Schema,
 } from '@aws-amplify/datastore';
 
 import { DataStoreActionFields, isAlreadyTyped } from './types';
 
-interface UseTypeCastFieldsProps<Model> {
+interface UseTypeCastFieldsProps<Model extends PersistentModel> {
   fields: DataStoreActionFields<Model>;
   modelName: string;
-  schema: Schema;
+  schema?: Schema;
```

Comment:
@reesscot Does loosening our types have any impact on Studio?

Reply:
We probably need to have this published to a tag and run the studio integ tests

Reply:
We should combine this and #3382 into a branch and publish a tag to run studio integ tests.

Reply:
Agreed, let's get a tag release out for testing on our end. Should be a quick yes/no outcome since only types are affected.